common/src/main/templates/io/netty/util/collection/KObjectHashMap.template

```diff
@@ -350,9 +350,10 @@ private int indexOf(int key) {
     /**
      * Returns the hashed index for the given key.
      */
-    private int hashIndex(int key) {
+    private int hashIndex(@k@ key) {
         // Allowing for negative keys by adding the length after the first mod operation.
-        return (key % keys.length + keys.length) % keys.length;
+        int hashCode = @HASH_CODE@;
```

Comment:
@Scottmitch I think I misunderstood what you meant.  I had this in a separate method before, which I changed to being "inline" :)

The problem is that every type except Long uses an identity hashcode. Long is a special case (see the pom.xml).  Maybe that logic will work for all types (I haven't thought about it)?